### DISCUSSION TITLE
Error "The mismatch is caused by single bit flip in data block at byte 37672, bit 7"

### DISCUSSION THREAD

**Discussion by Wachynaky** [ClickHouse/ClickHouse#54202] [Q&A] [answered]
Hello! I run this query to a table with 1.5 billions rows and 90 GB size

```

SELECT 
    date,
    count() "number of new URLs",
    arrayStringConcat(groupArray(url),', ') "URLs"
FROM
    (
    SELECT
        url,
        min(data_date) date
    FROM
        nameDatabase.nameTable
    where
        data_date > today() - 90
    group by
        url
)
GROUP BY
    date
ORDER BY
    date;

```
But CH shows me this message
`Code: 40. DB::Exception: Received from localhost:9000. DB::Exception: Checksum doesn't match: corrupted data. Reference: 291c8da497470154879e0f00d10660ff. Actual: 2dedaa17f0429a6d327d4ba0417bde09. Size of compressed block: 226228. The mismatch is caused by single bit flip in data block at byte 37672, bit 7. This is most likely due to hardware failure. If you receive broken data over network and the error does not repeat every time, this can be caused by bad RAM on network interface controller or bad controller itself or bad RAM on network switches or bad CPU on network switches (look at the logs on related network switches; note that TCP checksums don't help) or bad RAM on host (look at dmesg or kern.log for enormous amount of EDAC errors, ECC-related reports, Machine Check Exceptions, mcelog; note that ECC memory can fail if the number of errors is huge) or bad CPU on host. If you read data from disk, this can be caused by disk bit rot. This exception protects ClickHouse from data corruption due to hardware failures.: (while reading column url): (while reading from part /var/lib/clickhouse/store/89d/89d52328-3865-4355-8d75-62e0b7916014/202307_601_1302_3/ in table Database.nameTable (89d52328-3865-4355-8d75-62e0b7916014) located on disk default of type local, from mark 13956 with max_rows_to_read = 8192): While executing MergeTreeThread. (CHECKSUM_DOESNT_MATCH)`

Any idea to solve?
Thanks!

**Comment by Wachynaky** [comment]
The same query but using external data ("from  file('/media/my-local-buckets/*','CSVWithNames'....")  works! 

I'll try import one more time all data to a new table

**Reply by UnamedRus** [reply, marked as answer]
Check your disks/RAM

https://github.com/marliotto/clickhouse-bitflip
